src/main/scala/async/Async.scala

```diff
@@ -180,78 +178,44 @@ object Async:
           it.next.poll(listener)
```

Comment:
Actually, I think that we might drop wrapping the listener completely since we adapted the new poll policy because it does exactly what we want. The only problem I see is that it might confuse k to receive the upstream source instead of the race source.
```suggestion
          found = it.next.poll(k)
```
I guess that we just deprive the `listener`-of-`k` wrapping of any logic and it only forwards every request, substituting the source where necessary.

Reply:
That makes sense!